lib/graphite_async.ml

```diff
@@ -111,6 +112,35 @@ let send t () =
           { path=key; value=data; ts=now } :: acc))
   |> send_raw t
 
+module Report = struct
+  type metric = {
+    observations: int;
+    rate: float;
+  } [@@deriving show]
+
+  type t = (string * metric) list [@@deriving show]
+end
+
+let generate_report t =
```

Comment:
I think this can be just called `report`, since the signature `t -> Report.t` is pretty straightforward. `report` is also a verb so…